File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/utils/serialization/ValueTree.java
package net.namekdev.entity_tracker.utils.serialization;

public class ValueTree {
	public Object[] values;
	public ObjectModelNode model;
	public ValueTree parent;

	public ValueTree() {
	}

	public ValueTree(int length) {
		values = new Object[length];
	}
}

File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/utils/serialization/ValueTree.kt
package net.namekdev.entity_tracker.utils.serialization

class ValueTree {
    var id: Short = -1
    val values: Array<Any?>
    var model: ObjectModelNode? = null
    var parent: ValueTree? = null

    constructor(length: Int) {
        values = arrayOfNulls<Any>(length)
    }

    constructor(values: Array<Any?>) {
        this.values = values
    }
}
